components/dashboard/src/repositories/detail/workspaces/ConfigurationWorkspaceClassesOptions.tsx

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2024 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import type { Configuration } from "@gitpod/public-api/lib/gitpod/v1/configuration_pb";
+import { useState } from "react";
+import { Heading3, Subheading } from "@podkit/typography/Headings";
+import { ConfigurationSettingsField } from "../ConfigurationSettingsField";
+import { useConfigurationMutation } from "../../../data/configurations/configuration-queries";
+import { useAllowedWorkspaceClassesMemo } from "../../../data/workspaces/workspace-classes-query";
+import { Button } from "@podkit/buttons/Button";
+import { AlertTriangleIcon } from "lucide-react";
+import { useMutation } from "@tanstack/react-query";
+import {
+    WorkspaceClassesModifyModal,
+    WorkspaceClassesModifyModalProps,
+    WorkspaceClassesOptions,
+} from "../../../components/WorkspaceClassesOptions";
+
+export const ConfigurationWorkspaceClassesOptions = ({ configuration }: { configuration: Configuration }) => {
+    const [showModal, setShowModal] = useState(false);
+    const configurationMutation = useConfigurationMutation();
+
+    const { data: allowedClassesInConfiguration } = useAllowedWorkspaceClassesMemo(configuration.id, {
+        filterOutDisabled: true,
+    });
+    const { data: allowedClassesInOrganization } = useAllowedWorkspaceClassesMemo(undefined, {
+        filterOutDisabled: false,
+        ignoreScope: ["configuration"],
+    });
+
+    const updateMutation: WorkspaceClassesModifyModalProps["updateMutation"] = useMutation({
+        mutationFn: async ({ restrictedWorkspaceClasses, defaultClass }) => {
+            await configurationMutation.mutateAsync({
+                configurationId: configuration.id,
+                workspaceSettings: {
+                    workspaceClass: defaultClass,
+                    restrictedWorkspaceClasses,
+                    updateRestrictedWorkspaceClasses: true,
+                },
+            });
+        },
+    });
```

Comment:
Perhaps worth extracting to configuration queries instead of keeping in the component.

Reply:
We do it on purpose. This one is only generated for `WorkspaceClassModifyModal` only. And `WorkspaceClassModifyModal` is built for re-use.